peers/app_request_network.go

```diff
@@ -111,7 +113,8 @@ type appRequestNetwork struct {
 	lruSubnets         *linked.Hashmap[ids.ID, interface{}]
 	trackedSubnetsLock *sync.RWMutex
 
-	latestSyncedPChainHeight uint64
+	latestSyncedPChainHeight     uint64
```

Comment:
Can we use an atomic int for this?

Reply:
Ah, gotcha. Yeah it'd need to be:
```
for {
    lastSynced := n.latestSyncedPChainHeightLock.Load()
    if pChainHeight <= lastSynced {
        return
    }
    if n.latestSyncedPChainHeightLock.CompareAndSwap(lastSyned, pChainHeight) {
        return
    }
}
```

Reply:
I do think avoid explicit mutex's is best practice in general since it's hard to mess up in the future (either causing deadlock or race conditions)